### PR TITLE
internal/prometheus: use jsonpb for json marshalling

### PR DESCRIPTION
**Pull request by srenatus**
We're now using jsonpb for the prometheus logs.

### CODE REVIEW

File: internal/prometheus/prometheus.go [hunk 1]
```diff
@@ -116,12 +120,21 @@ func (p *Provider) All() map[string]interface{} {
...
 	}
 
 	return all
 }
 
+type wrap struct{ family proto.Message }
+
+var marshaler = jsonpb.Marshaler{}
+
+func (w wrap) MarshalJSON() ([]byte, error) {
```
Comment (srenatus) [inline internal/prometheus/prometheus.go:133]: trying to have the reflect machinery underneath encoding/json pick up that these metrics should use a different marshalling function to generate json.

File: internal/prometheus/prometheus_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	}(logging.NewNoOpLogger())
+
+	prom := New(inner, logger)
+
+	m := prom.All()
+	bs, err := json.Marshal(m)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Log(string(bs))
```
Comment (ashutosh-narkar) [inline internal/prometheus/prometheus_test.go:30]: Do we need to verify something here ?

Comment (srenatus) [reply internal/prometheus/prometheus_test.go:30]: We could expand on it for sure. Right now, it tells us that what happened when we've bumped github.com/prometheus/client_golang the last time won't happen again 😅

**Comment by srenatus** [comment]
We'll also have to update the Status API docs, https://www.openpolicyagent.org/docs/edge/management-status/. Moving to draft.

**Comment by tsandall** [comment]
We're going to need a big release note for this one right? This breaks anyone that's using the prometheus metrics in the Status API? I'm guessing that those are not widely used but it'll still have to get communicated.

**Comment by srenatus** [comment]
> We're going to need a big release note for this one right?

Certainly! I'll add the CHANGELOG note in this PR.

**Comment by srenatus** [comment]
- [x] added tests
- [x] updated docs
- [x] added changelog item on breaking change

**Review by ashutosh-narkar — APPROVED** [review]
LGTM

**Pull request merged**